mods/gen6/random-teams.js

```diff
@@ -540,7 +540,7 @@ class RandomGen6Teams extends RandomTeams {
 
 				// Remove rejected moves from the move list
 				if (rejected && (movePool.length - availableHP || availableHP && (moveid === 'hiddenpower' || !hasMove['hiddenpower']))) {
-					moves.splice(k, 1);
+					moves.splice(moves.indexOf(move), 1);
```

Comment:
Literally never _add_ an `indexOf`, even if it means you have to stick to an old-style loop.

Reply:
(`i`)